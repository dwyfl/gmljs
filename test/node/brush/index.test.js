import GML from '../../../src';
import GMLBrush from '../../../src/node/brush';
import testXml from './data/test.xml';

it('creates a correct default GMLBrush node', () => {
  const gml = GMLBrush.create().toString();
  expect(gml).toMatchSnapshot();
});

it('creates a correct GMLBrush node from spec XML', () => {
  const gmlNode = GML.createNodeFromXml(GMLBrush, testXml);
  const gml = gmlNode.toString();
  expect(gml).toMatchSnapshot();
});