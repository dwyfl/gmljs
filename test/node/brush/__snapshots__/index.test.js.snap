// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a correct GMLBrush node from spec XML 1`] = `"<brush><mode>0</mode><uniquestyleid>LaserTagArrowLetters</uniquestyleid><spec>http://aurltodescribethebrushspec.com/someSpec.xml</spec><width>10</width><speedtowidthratio>1.5</speedtowidthratio><dripamnt>1</dripamnt><dripspeed>1</dripspeed><layerabsolute>0</layerabsolute><color><r>255</r><g>255</g><b>255</b><a>255</a></color><dripvecrelativetoup><x>0</x><y>1</y><z>0</z></dripvecrelativetoup></brush>"`;

exports[`creates a correct default GMLBrush node 1`] = `"<brush></brush>"`;
